src/vic3_world/world/vic3_world_importer.cpp

```diff
@@ -25,38 +27,83 @@
 namespace
 {
 
-std::istringstream MeltSave(std::string_view save_filename, bool debug)
+std::string ReadSave(std::string_view save_filename)
 {
    std::ifstream save_file(std::filesystem::u8path(save_filename), std::ios::in | std::ios::binary);
    const auto save_size = static_cast<std::streamsize>(std::filesystem::file_size(save_filename));
    std::string save_string(save_size, '\0');
    save_file.read(save_string.data(), save_size);
 
-   std::string liquid;
-   const auto save = rakaly::parseVic3(save_string);
+   return save_string;
+}
+
+
+std::vector<std::string> ReadModNames(const rakaly::GameFile& save, const std::string& save_string)
+{
+   std::string save_meta;
    if (save.is_binary())
    {
-      auto melt = save.melt();
-      if (melt.has_unknown_tokens())
+      const auto melt = save.meltMeta();
+      if (!melt || melt->has_unknown_tokens())
       {
-         throw std::runtime_error("Unable to melt ironman save");
+         throw std::runtime_error("Unable to melt ironman save's metadata");
       }
 
-      melt.writeData(liquid);
+      melt->writeData(save_meta);
    }
    else
    {
-      liquid = save_string;
+      save_meta = save_string;
    }
+   std::istringstream meta_stream{save_meta};
+
+   std::vector<std::string> mod_names;
+
+   commonItems::parser meta_parser;
+   meta_parser.registerKeyword("mods", [&mod_names](std::istream& input_stream) {
+      mod_names = commonItems::stringList(input_stream).getStrings();
+   });
+   meta_parser.IgnoreUnregisteredItems();
+   meta_parser.parseStream(meta_stream);
```

Comment:
meta is not guaranteed. I think. You should test it with debug mode so it exports a unified text save. You won't get a separate meta there I believe so may have to source mods from gamestate itself.

Reply:
Surprisingly it's there. I'm sure users will send me something broken in no time though.